tools/provision

```diff
@@ -1,33 +1,48 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
 BASEDIR=$(dirname `dirname $0`)
+VENVEXECDIR="bin"
+PYVERSION=$(python -V | grep -Po '(?<=Python ).+')
+PYCUSTOMPATH_ARG=""
+while getopts "hp:" arg; do
+    case "$arg" in
+        h)
+            echo "USAGE: $0 [-p <path_to_python_version>]"
+            exit
+            ;;
+        p)
+            PYCUSTOMPATH_ARG="-p $OPTARG"
+            PYVERSION=$("$OPTARG" "-V" 2>&1 | grep -Po '(?<=Python ).+')
+            ;;
+    esac
+done
+VENVDIR="zulip-api-py$PYVERSION-venv"
 
-if [ ! -d "$BASEDIR/zulip-api-py2-venv" ]; then
-    virtualenv $BASEDIR/zulip-api-py2-venv
+if [ ! -d "$BASEDIR/$VENVDIR" ]; then
+    virtualenv $PYCUSTOMPATH_ARG "$BASEDIR/$VENVDIR"
     echo "Virtualenv created."
 fi
 
-VENVBINDIR="bin"
-if [[ "$OSTYPE" == "msys" ]]; then
+if [[ ! -d "$BASEDIR/$VENVDIR/$VENVEXECDIR" ]]; then
```

Comment:
(The risk with this kind of test is that some other bug results in it being detected as Windows, causing its own problems, and then folks spend time debugging the wrong thing).